### ISSUE TITLE
Java version not compatible with Gradle

### ISSUE THREAD

**Issue by dmwelch** [AdoptOpenJDK/homebrew-openjdk#55]
I'm setting up Minecraft Forge to use AdoptOpenJDK from [here](https://mcforge.readthedocs.io/en/latest/gettingstarted/), but Gradle can't parse the Java version correctly:

> FAILURE: Build failed with an exception.
>
> * What went wrong:
> Could not determine java version from '10.0.2'

Changing my `JAVA_HOME` to point to Oracle's JDK (1.8) works fine but any AOJDK versions I have installed goes pearshaped.

**Comment by VonUniGE** [comment]
What is the result if you use AdoptOpenJDK 8?

**Comment by karianna** [comment]
@dmwelch And which version of AdoptOpenJDK are you using?  Can you run `java --version` and tell us the results?

**Comment by gdams** [comment]
ping @dmwelch

**Comment by gdams** [comment]
@dmwelch I think we've fixed this. Can you try with our adoptopenjdk12-openj9 cask to check?

**Comment by rafasoares** [comment]
Not OP, but I'm having the same issue:

```sh
Could not determine Java version using executable /Library/Java/JavaVirtualMachines/adoptopenjdk-11.0.2.jdk/Contents/Home/bin/java.
```

```sh
$ java -version
openjdk version "11.0.2" 2019-01-15
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.2+9)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.2+9, mixed mode)
```

I've just installed OpenJDK via `brew cask install adoptopenjdk`.  I'm going to try `adoptopenjdk12-openj9` and report back.

**Comment by douph1** [comment]
A gradle debug output could help to understand which gradle class check version numbering to know what is ok
Maybe https://github.com/gradle/gradle/blob/master/subprojects/core/src/main/java/org/gradle/util/VersionNumber.java ?
Wich gradle version ? as master  says:  * <p>This class considers missing parts to be 0, so that "1.0" == "1.0.0" == "1.0.0_0".</p>
but maybe the parsing problem is before the number ..

**Comment by douph1** [comment]
gradle :
./subprojects/base-services/src/main/java/org/gradle/api/JavaVersion.java
a gradle stack trace would be good

* Try:
Run with --stacktrace option to get the stack trace.

**Comment by rafasoares** [comment]
Here is the stack trace:
```sh
./gradlew setupDecompWorkspace --stacktrace --debug
21:06:58.832 [ERROR] [org.gradle.BuildExceptionReporter]
21:06:58.924 [ERROR] [org.gradle.BuildExceptionReporter] FAILURE: Build failed with an exception.
21:06:58.928 [ERROR] [org.gradle.BuildExceptionReporter]
21:06:58.928 [ERROR] [org.gradle.BuildExceptionReporter] * What went wrong:
21:06:58.928 [ERROR] [org.gradle.BuildExceptionReporter] Could not determine java version from '12'.
21:06:58.930 [ERROR] [org.gradle.BuildExceptionReporter]
21:06:58.931 [ERROR] [org.gradle.BuildExceptionReporter] * Exception is:
21:06:58.931 [ERROR] [org.gradle.BuildExceptionReporter] java.lang.IllegalArgumentException: Could not determine java version from '12'.
21:06:58.932 [ERROR] [org.gradle.BuildExceptionReporter]        at org.gradle.api.JavaVersion.toVersion(JavaVersion.java:68)
21:06:58.932 [ERROR] [org.gradle.BuildExceptionReporter]        at org.gradle.api.JavaVersion.current(JavaVersion.java:78)
21:06:58.932 [ERROR] [org.gradle.BuildExceptionReporter]        at org.gradle.launcher.cli.JavaRuntimeValidationAction.execute(JavaRuntimeValidationAction.java:32)
21:06:58.932 [ERROR] [org.gradle.BuildExceptionReporter]        at org.gradle.launcher.cli.JavaRuntimeValidationAction.execute(JavaRuntimeValidationAction.java:24)
21:06:58.932 [ERROR] [org.gradle.BuildExceptionReporter]        at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:33)
21:06:58.933 [ERROR] [org.gradle.BuildExceptionReporter]        at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:22)
21:06:58.933 [ERROR] [org.gradle.BuildExceptionReporter]        at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:205)
21:06:58.933 [ERROR] [org.gradle.BuildExceptionReporter]        at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:169)
21:06:58.933 [ERROR] [org.gradle.BuildExceptionReporter]        at org.gradle.launcher.Main.doAction(Main.java:33)
21:06:58.933 [ERROR] [org.gradle.BuildExceptionReporter]        at org.gradle.launcher.bootstrap.EntryPoint.run(EntryPoint.java:45)
21:06:58.933 [ERROR] [org.gradle.BuildExceptionReporter]        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
21:06:58.934 [ERROR] [org.gradle.BuildExceptionReporter]        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
21:06:58.934 [ERROR] [org.gradle.BuildExceptionReporter]        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
21:06:58.934 [ERROR] [org.gradle.BuildExceptionReporter]        at java.base/java.lang.reflect.Method.invoke(Method.java:567)
21:06:58.934 [ERROR] [org.gradle.BuildExceptionReporter]        at org.gradle.launcher.bootstrap.ProcessBootstrap.runNoExit(ProcessBootstrap.java:55)
21:06:58.934 [ERROR] [org.gradle.BuildExceptionReporter]        at org.gradle.launcher.bootstrap.ProcessBootstrap.run(ProcessBootstrap.java:36)
21:06:58.935 [ERROR] [org.gradle.BuildExceptionReporter]        at org.gradle.launcher.GradleMain.main(GradleMain.java:23)
21:06:58.935 [ERROR] [org.gradle.BuildExceptionReporter]        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
21:06:58.935 [ERROR] [org.gradle.BuildExceptionReporter]        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
21:06:58.935 [ERROR] [org.gradle.BuildExceptionReporter]        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
21:06:58.935 [ERROR] [org.gradle.BuildExceptionReporter]        at java.base/java.lang.reflect.Method.invoke(Method.java:567)
21:06:58.935 [ERROR] [org.gradle.BuildExceptionReporter]        at org.gradle.wrapper.BootstrapMainStarter.start(BootstrapMainStarter.java:30)
21:06:58.936 [ERROR] [org.gradle.BuildExceptionReporter]        at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:127)
21:06:58.936 [ERROR] [org.gradle.BuildExceptionReporter]        at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:61)
```

One thing to note: While debugging another issue with Forge, someone asked to check the Java version with `javac -version`. The output of that command with AdoptOpenJDK is `javac 12`, while with the `java8` cask, it's `javac 1.8.0_202`. I'm guessing this is somewhat related.

**Comment by douph1** [comment]
@rafasoares can you point us to which gradle version you use ?

**Comment by douph1** [comment]
>There are a few issues with Gradle 5.0-rc-1, but in general, it works with 4.10.2

**Comment by karianna** [comment]
Latest is 5.3.1 works for me.

**Comment by rafasoares** [comment]
It's actually the opposite problem, I've got the error with an older version of Gradle, 2.14 - the one provided with the latest stable Forge version.

I've started a blank Gradle project, which uses 4.10.3, and manually set up the Forge dependencies and it worked (with both `adoptopenjdk` and `adoptopenjdk12-openj9`).

I'll take this up with the Forge team, thanks!

**Comment by ItsCompileTime** [comment]
Receiving a similar error with the Gradle version 5.6.3

* What went wrong:
Execution failed for task ':run'.
> Process 'command '/Library/Java/JavaVirtualMachines/adoptopenjdk-12-openj9.jdk/Contents/Home/bin/java'' finished with non-zero exit value 1


Specifications
------------------------------------------------------------
Gradle 5.6.2
------------------------------------------------------------

Build time:   2019-09-05 16:13:54 UTC
Revision:     55a5e53d855db8fc7b0e494412fc624051a8e781

Kotlin:       1.3.41
Groovy:       2.5.4
Ant:          Apache Ant(TM) version 1.9.14 compiled on March 12 2019
JVM:          12.0.2 (Eclipse OpenJ9 openj9-0.15.1)
OS:           Mac OS X 10.13.6 x86_64

**Comment by aahlenst** [comment]
Closing due to inactivity. If this problem pops up again, please open a new issue.

**Comment by Naruto-uchiha24** [comment]
![image](https://user-images.githubusercontent.com/73196181/97085556-cb9ede00-163b-11eb-85ea-9bf08499671d.png)
can anyone tell me what's problem with mine

**Comment by Naruto-uchiha24** [comment]
anyone please help @karianna @aahlenst @rafasoares anyone.....

**Comment by aahlenst** [comment]
Gradle 4.4.1 is ancient. [The current version is 6.7](https://gradle.org/releases/). Update and you should be good.

**Comment by Jeeppler** [comment]
I tested it an a Debian 10 container with Gradle 6.7.1 and AdoptOpenJDK 15 OpenJ9. It crashed immediately with a JVM dump.

~~~
./gradlew tasks
Unhandled exception
Type=Segmentation error vmState=0x00000000
…
~~~

Java Version:

~~~
java --version
openjdk 15.0.1 2020-10-20
OpenJDK Runtime Environment AdoptOpenJDK (build 15.0.1+9)
Eclipse OpenJ9 VM AdoptOpenJDK (build openj9-0.23.0, JRE 15 Linux amd64-64-Bit Compressed References 20201022_81 (JIT enabled, AOT enabled)
OpenJ9   - 0394ef754
OMR      - 582366ae5
JCL      - ad583de3b5 based on jdk-15.0.1+9)
~~~

I did the same test with AdoptOpenJDK 15 Hotspot and everything worked out of the box in the same environment.

This issue might be in the wrong repository as this issue is not related to `hombrew-opendjdk`, but I do not know where else to post it.
